packages/venia-ui/lib/RootComponents/Category/categoryContent.js

```diff
@@ -30,80 +38,74 @@ const CategoryContent = props => {
         categoryName,
         categoryDescription,
         filters,
-        handleLoadFilters,
-        handleOpenFilters,
         items,
         totalPagesFromData
     } = talonProps;
 
     const classes = mergeClasses(defaultClasses, props.classes);
 
-    const maybeFilterButtons = filters ? (
-        <Button
-            priority={'low'}
-            classes={{ root_lowPriority: classes.filterButton }}
-            onClick={handleOpenFilters}
-            onFocus={handleLoadFilters}
-            onMouseOver={handleLoadFilters}
-            type="button"
-        >
-            <FormattedMessage
-                id={'categoryContent.filter'}
-                defaultMessage={'Filter'}
-            />
-        </Button>
+    const maybeFilterButtons = filters && filters.length ? (
+        <FilterModalOpenButton filters={filters} />
     ) : null;
 
+    const filtersModal = filters && filters.length ? (
+        <FilterModal filters={filters} />
+    ) : null;
+
+    const filtersAndTotalPagesPresent = !!(totalPagesFromData && filters);
+
     const maybeSortButton =
-        totalPagesFromData && filters ? (
+        filtersAndTotalPagesPresent ? (
             <ProductSort sortProps={sortProps} />
         ) : null;
 
-    const maybeSortContainer =
-        totalPagesFromData && filters ? (
-            <div className={classes.sortContainer}>
-                <FormattedMessage
-                    id={'categoryContent.itemsSortedBy'}
-                    defaultMessage={'Items sorted by '}
-                />
-                <span className={classes.sortText}>
-                    <FormattedMessage
-                        id={currentSort.sortId}
-                        defaultMessage={currentSort.sortText}
-                    />
-                </span>
-            </div>
-        ) : null;
-
-    // If you want to defer the loading of the FilterModal until user interaction
-    // (hover, focus, click), simply add the talon's `loadFilters` prop as
-    // part of the conditional here.
-    const modal = filters ? <FilterModal filters={filters} /> : null;
+    const maybeSortContainer = <SortedByContainer
+        shouldDisplay={filtersAndTotalPagesPresent}
+        currentSort={currentSort}
+    />;
```

Comment:
The pattern we use in the repo is to do a conditional render rather than a prop used by the component. This is just a choice to leave render responsibility to the parent. Please swap this for the original pattern.

```suggestion
    const maybeSortContainer = filtersAndTotalPagesPresent ? <SortedByContainer
        currentSort={currentSort}
    /> : null;
```